libs/velo-external-db-core/src/service/schema.ts

```diff
@@ -100,6 +101,17 @@ export default class SchemaService {
             throw new errors.UnsupportedSchemaOperation(`Your database doesn't support ${operationName} operation`, collectionName, operationName)
     }
 
+    private async validateFields(fields: collectionSpi.Field[]) {
+        if (!fields.length) {
+            throw new errors.InvalidRequest('Fields are missing')
+        }
+
+        const fieldsName = fields.map(field => field.key)
+        if (!fieldsName.includes(PrimaryKeyFieldName)) {
+            throw new errors.InvalidRequest(`${PrimaryKeyFieldName} is field is missing`)
+        }
+    }
+
```

Comment:
1. it's not async.
2. the first check is redundant - it would succeed only if there's _id column, so you don't need to check if it's empty before it.

Reply:
and maybe `${PrimaryKeyFieldName} is field is missing` --> `${PrimaryKeyFieldName} field is missing`